src/dragonball/src/dbs_upcall/src/lib.rs

```diff
@@ -275,18 +303,29 @@ pub struct UpcallEpollHandler<S: UpcallClientService + Send> {
     reconnect_timer: TimerFd,
     reconnect_time: u32,
     in_reconnect: bool,
+    metrics: Arc<UpcallClientMetrics>,
 }
 
 impl<S: UpcallClientService + Send> UpcallEpollHandler<S> {
-    fn new(info: Arc<Mutex<UpcallClientInfo<S>>>) -> Result<Self> {
+    fn new(
+        info: Arc<Mutex<UpcallClientInfo<S>>>,
+        metrics: Arc<UpcallClientMetrics>,
+    ) -> Result<Self> {
         let handler = UpcallEpollHandler {
             info,
             reconnect_timer: TimerFd::new().map_err(UpcallClientError::TimerFd)?,
             reconnect_time: 0,
             in_reconnect: false,
+            metrics: metrics.clone(),
         };
         let info = handler.info.clone();
-        info.lock().unwrap().server_connection_start()?;
+        info.lock()
+            .unwrap()
+            .server_connection_start()
+            .map_err(|e| {
+                metrics.connect_start_fails.inc();
```

Comment:
there are multiple places using `server_connection_start()` but related metric only count here which is not enough.

How about add this metric inc() into the function `server_connection_start`?

Reply:
Agree with you. I fixed it.

Reply:
I think the count of upcall request count is helpful, we can add this to metrics.

Reply:
The metric of upcall request counts was added to metrics.